### PR TITLE
do not adding any non-global addresses to cache anymore

### PR DESCRIPTION
**Pull request by vinipsmaker**
Updates #388

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/crust/402)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @inetic

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by vinipsmaker** [comment]
`service::test::bootstrap` fails after I filter out non-global addresses from the list of cached addresses.

After investigating the problem, here are my findings:

- The test start two `Service` objects, bootstraping both and hoping they'll find each other through beacon. Obvious.
- They DO find each other and a connection IS established.
- Different from normal connection establishment, the "connecting" side (function `state::seek_peers`) will actually drop the `Transport` object after receiving the first message, just to add the list of endpoints received in this first message to the list of endpoints to be attempted in the bootstrap attempt.
- By this time, the `BootstrapFinished` event will be already emitted.
- So, we've done a successful connection establishment with handshake and all neat stuff, just to change the state of `Service` to `BootstrapFinished` and then drop the "good" connection that we already had established.
- Beacon handling must be reworked.

**Comment by vinipsmaker** [comment]
just in case the new build succeed by luck, here is the previous failing test: https://ci.appveyor.com/project/MaidSafe-QA/crust/build/1.0.374/job/a63lrqvf2rsho130#L892

**Comment by vinipsmaker** [comment]
tests are passing, PR is ready

File: src/util.rs [hunk 1]
```diff
@@ -355,7 +355,7 @@ pub fn loopback_if_unspecified(addr : IpAddr) -> IpAddr {
 pub fn random_endpoint() -> ::transport::Endpoint {
     // TODO - randomise V4/V6 and TCP/UTP
     let address = ::std::net::SocketAddrV4::new(
-        ::std::net::Ipv4Addr::new(::rand::random::<u8>(),
+        ::std::net::Ipv4Addr::new(173, // ensure is a global addr
                                   ::rand::random::<u8>(),
                                   ::rand::random::<u8>(),
                                   ::rand::random::<u8>()),
```
Comment (inetic) [inline src/util.rs]: I think we should rename this function to `random_global_endpoint`.

Comment (vinipsmaker) [inline src/util.rs]: Done

**Pull request merged**